zkchannels-contract/zkchannel_smartpy_script.py

```diff
@@ -40,89 +39,113 @@ def is_g1_not_zero(val):
-            sp.verify(self.data.merchBal == sp.tez(0))
+            # Set merchBal to the merchant's funding amount. 
             self.data.merchBal = self.data.merchFunding
-        # If cust and merch Balances have been funded, mark the channel as open.
+
+        # If cust and merch Balances have been funded, set the contract status to OPEN.
         sp.if ((self.data.custBal == self.data.custFunding) & (self.data.merchBal == self.data.merchFunding)):
+            # Set channel status to OPEN
             self.data.status = OPEN
 
-
-    # reclaimFunding allows the customer or merchant to withdraw funds
-    # if the other party has not funded their side of the channel yet.
+    # reclaimFunding allows the customer to withdraw their funds
+    # if the merchant has not funded their side of the channel yet.
     @sp.entry_point
     def reclaimFunding(self):
-        sp.verify(self.data.status == AWAITING_FUNDING)
-        sp.if self.data.custAddr == sp.sender:
-            sp.verify(self.data.custBal == self.data.custFunding)
-            sp.send(self.data.custAddr, self.data.custBal)
-            self.data.custBal = sp.tez(0)
-        sp.if self.data.merchAddr == sp.sender:
-            sp.verify(self.data.merchBal == self.data.merchFunding)
-            sp.send(self.data.merchAddr, self.data.merchBal)
-            self.data.merchBal = sp.tez(0)
+        # Verify that the channel is still AWAITING_MERCH_FUNDING.
+        sp.verify(self.data.status == AWAITING_MERCH_FUNDING)
+        # Only allow the customer to call the entrypoint.
+        sp.verify(self.data.custAddr == sp.sender)
+        # Verify that they had funded their side of the channel.
+        sp.verify(self.data.custBal == self.data.custFunding)
+        # Send the customer’s balance back to custAddr.
+        sp.send(self.data.custAddr, self.data.custBal)
+        # Set the channel status to FUNDING_RECLAIMED.
+        self.data.status = FUNDING_RECLAIMED
  
     # expiry can be called by the merchant to initiate channel closure.
     # The customer should call custClose using the latest state. Otherwise,
     # after the delay expires, the merchant will be able to claim all the
     # funds in the channel using merchClaim.
     @sp.entry_point
     def expiry(self):
+        # Only allow the merchant to call the entrypoint.
         sp.verify(self.data.merchAddr == sp.sender)
+        # Verify that the contract status is OPEN.
         sp.verify(self.data.status == OPEN)
+        # Record the time when the delay period will expire with self.data.delayExpiry.
+        # The time is calculated by adding the number of seconds in selfDelay to timestamp of
+        # the block it is confirmed in.
         self.data.delayExpiry = sp.now.add_seconds(self.data.selfDelay)
         self.data.status = EXPIRY
  
     # merchClaim can be called by the merchant if the customer has not called
     # custClose before the delay period has expired.
```

Comment:
Clarify that this is only callable if the `expiry` entrypoint has already been called.

Reply:
also added that the delay period needs to have passed:
```
    # merchClaim can only be called by the merchant after the expiry entrypoint has been called and # the delay period defined by selfDelay has elapsed. If the customer calls custClose, merchClaim cannot be called.
````